src/main/java/de/tum/in/www1/artemis/web/rest/dto/standardizedCompetency/KnowledgeAreaRequestDTO.java

```diff
@@ -11,7 +12,7 @@
  * DTO used to send requests regarding {@link KnowledgeArea} objects. It has no id and only contains the id (not object) of its parent
  */
 @JsonInclude(JsonInclude.Include.NON_EMPTY)
-public record KnowledgeAreaRequestDTO(@NotNull @Size(min = 1, max = KnowledgeArea.MAX_TITLE_LENGTH) String title,
-        @NotNull @Size(min = 1, max = KnowledgeArea.MAX_SHORT_TITLE_LENGTH) String shortTitle, @Size(max = KnowledgeArea.MAX_DESCRIPTION_LENGTH) String description,
+public record KnowledgeAreaRequestDTO(@NonNull @Size(min = 1, max = KnowledgeArea.MAX_TITLE_LENGTH) String title,
+        @NonNull @Size(min = 1, max = KnowledgeArea.MAX_SHORT_TITLE_LENGTH) String shortTitle, @Size(max = KnowledgeArea.MAX_DESCRIPTION_LENGTH) String description,
         Long parentId) {
```

Comment:
This seems to no longer reject `null` as title / shortTitle values, causing a 500 instead of a 400 in the tests.

```
MockHttpServletResponse:
           Status = 500
    Error message = null
          Headers = [Vary:"Origin", "Access-Control-Request-Method", "Access-Control-Request-Headers", Content-Version:"1.3.3-beta7", Content-Type:"application/problem+json", X-Content-Type-Options:"nosniff", X-XSS-Protection:"0", Cache-Control:"no-cache, no-store, max-age=0, must-revalidate", Pragma:"no-cache", Expires:"0", Content-Security-Policy:"script-src 'self' 'unsafe-inline' 'unsafe-eval'", Referrer-Policy:"strict-origin-when-cross-origin", Permissions-Policy:"camera=(), fullscreen=(*), geolocation=(), gyroscope=(), magnetometer=(), microphone=(), midi=(), payment=(), sync-xhr=()"]
     Content type = application/problem+json
             Body = {
  "type" : "https://www.jhipster.tech/problem/problem-with-message",
  "title" : "Internal Server Error",
  "status" : 500,
  "detail" : "could not execute statement [NULL not allowed for column \"TITLE\"; SQL statement:\ninsert into knowledge_area (description,parent_id,short_title,title,id) values (?,?,?,?,default) [23502-224]] [insert into knowledge_area (description,parent_id,short_title,title,id) values (?,?,?,?,default)]; SQL [insert into knowledge_area (description,parent_id,short_title,title,id) values (?,?,?,?,default)]; constraint [null]",
  "path" : "/api/admin/standardized-competencies/knowledge-areas",
  "message" : "error.http.500"
}
```

Reply:
I guess JSpecify is not directly supported by the checker that is run before endpoints are called. If there is no way to fix this, this would be a huge downside to using them IMO, as these automatic checks are rather beneficial.